File: VRP.Dtos/DataSetPredictModel.cs
using VRP.Core.Interfaces;

namespace VRP.Dtos {
    public class DataSetPredictModel : IId<long>, IDto {
        public long Id { get; set; }
        public double Target { get; set; }
        public double Predict { get; set; }
        public long DataSetId { get; set; }
    }
}

File: Client/src/app/data-sets/shared/models/data-set-predict.model.ts
export class DataSetPredictModel {
    id: number;
    dataSetId: number;
    target: number;
    predict: number;


    public getAbsolute(): number {
        return this.predict - this.target;
    }

    public getRelative(): number {
        return Math.abs(this.predict - this.target) / this.target;
    }

    public getPercentageDifference(): number {
        return Math.abs((this.predict - this.target) / ((this.predict + this.target) / 2)) * 100;
    }

    public constructor(fields?: Partial<DataSetPredictModel>) {

        if (fields) {
            Object.assign(this, fields);
        }
    }
}
